sanic/mixins/listeners.py

```diff
@@ -59,6 +82,8 @@ def register_listener(listener, event):
             return listener
 
         if callable(listener_or_event):
+            if TYPE_CHECKING:
+                assert event_or_none is not None
```

Comment:
Avoid `assert` in production code. There is another instance in the codebase where we have this, but we should not proliferate the issue. Use `raise` instead.